cluster-autoscaler/expander/priority/priority.go

```diff
@@ -135,8 +135,10 @@ func (p *priority) BestOption(expansionOptions []expander.Option, nodeInfo map[s
 	for _, option := range expansionOptions {
 		id := option.NodeGroup.Id()
 		found := false
+		skipped := false
 		for prio, nameRegexpList := range priorities {
 			if prio < maxPrio {
+				skipped = true
```

Comment:
If i understand the code correctly, this can only be set if maxPrio != -1, and when it happens found=true. So we have the following combinations possible:

found = false skipped = false -> log should be printed
found = true skipped = false -> log should not be printed 
found = true skipped = true -> log should not be printed

found = false and skipped=true is not possible

thus, line 154 can simply be left as it was and skipped (and the whole PR) is not needed

Reply:
`if prio > maxPrio {
				maxPrio = prio
				best = nil
			}`
 when it finds a higher priority, it updates maxPrio, so the node group with lower priority will be skipped.
But when skipped, it warns "node group %s not found in priority expander configuration.". It confuses customers. 

I change the conditions to: when the node group is not skipped and not matched, it throws warnings.

Reply:
Here is a example.
suppose the priority list is [[1, "\*"], [10, "A"], [11, "C"]], the nodegroup list is [Node("A"), Node("B")]. Node("A") will be the best option. Node("B") matches [1, "*"] but it still warns "**node group B not found in priority expander configuration.**"

Reply:
You are right, I fixed it in a wrong way. But the problem is still there. Here is an example.

Suppose the priority list is [[1, "\*"], [10, "A"], [11, "C"]], the nodegroup list is [Node("A"), Node("B")]. Node("A") will be the best option. Node("B") matches [1, "*"] but it warns "node group B not found in priority expander configuration."

Could we remove the block `if prio < maxPrio {
				continue
			}`. Yes it has some performance cost but we have to compare all the regex in order to warn misconfiguration to customers.